pysteps/nowcasts/lagrangian_probability.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+pysteps.nowcasts.lagrangian_probability
+======================================
+
+Implementation of Lagrangian probability nowcasting methods.
+
+.. autosummary::
+    :toctree: ../generated/
+
+    forecast
+"""
+
+import numpy as np
+from scipy import signal
+
+from pysteps.nowcasts import extrapolation
+
+
+def forecast(
+    precip,
+    velocity,
+    timesteps,
+    threshold,
+    extrap_method="semilagrangian",
+    extrap_kwargs=None,
+    slope=5,
+):
+    """
+    Generate a probability nowcast by a local lagrangian approach.
+
+    Parameters
+    ----------
+    precip: array-like
+       Two-dimensional array of shape (m,n) containing the input precipitation
+       field.
+    velocity: array-like
+       Array of shape (2,m,n) containing the x- and y-components of the
+       advection field. The velocities are assumed to represent one time step
+       between the inputs.
+    timesteps: int or list of floats
+       Number of time steps to forecast or a list of time steps for which the
+       forecasts are computed (relative to the input time step). The elements of
+       the list are required to be in ascending order.
+    threshold: float
+       Intensity threshold for which the exceedance probabilities are computed.
+    slope: float, optional
+       The slope of the relationship between optimum scale and lead time in
+       pixel / timesteps.
+
+    Returns
+    -------
+    out: ndarray_
+        Three-dimensional array of shape (num_timesteps, m, n) containing a time
+        series of nowcast exceedence probabilities. The time series starts from
+        t0 + timestep, where timestep is taken from the advection field velocity.
+
+    References
+    ----------
+    Germann, U. and I. Zawadzki, 2004:
+    Scale Dependence of the Predictability of Precipitation from Continental
+    Radar Images. Part II: Probability Forecasts.
+    Journal of Applied Meteorology, 43(1), 74-89.
+    """
+    if isinstance(timesteps, int):
+        timesteps = np.arange(1, timesteps + 1)
+    precip_forecast = extrapolation.forecast(
+        precip,
+        velocity,
+        timesteps,
+        extrap_method,
+        extrap_kwargs,
+    )
+    nanmask = np.isnan(precip_forecast)
+    precip_forecast[nanmask] = np.nanmin(precip_forecast)
+    precip_forecast = (precip_forecast > threshold).astype(float)
```

Comment:
mmmh not sure if there is a clear definition anywhere, but to me the word "exceed" translates more to "greater than" rather than "equal or greater than", no?

Reply:
done in caf4f8d

Reply:
Ok good. I was thinking in terms of warnings. 
Users activate when the "psychological" threshold is reached :-)